lib/mix/lib/mix/rebar.ex

```diff
@@ -151,6 +151,21 @@ defmodule Mix.Rebar do
   defp parse_source(source) do
     [scm, url | source] = Tuple.to_list(source)
 
+    {scm, additional_opts} =
+      case scm do
+        :git_subdir ->
+          case source do
+            [_, sparse_dir | _] ->
+              {:git, [{:sparse, sparse_dir}]}
+
+            _ ->
+              {:git, []}
+          end
+
+        _ ->
+          {:git, []}
+      end
```

Comment:
```suggestion
    {scm, additional_opts} =
      case {scm, source} do
        {:git_subdir, [_, sparse_dir | _]} -> {:git, [{:sparse, sparse_dir}]}
        {_, _} -> {:git, []}
      end
```

Reply:
Also, couldn't someone try to use git_subdir with branch/tag/ref? In this case, shouldn't this case also return the new source? Something like this:

```suggestion
    {scm, additional_opts, source} =
      case {scm, source} do
        {:git_subdir, [_, sparse_dir | source]} -> {:git, [sparse: sparse_dir], source}
        {_, _} -> {:git, [], source}
      end
```

?

Reply:
It has to include a branch/tag/ref, that is why the head of the source is skipped and then parsed in the next `case`.

Reply:
Format is: `{git_subdir, Url, BranchTagRef, Dir}`.

Reply:
ahhhhhhhhhhh, I see! So ignore my second suggestion, although I think my first suggestion is still good, right?